internal/provisioner/equality/equality.go

```diff
@@ -194,6 +194,7 @@ func NodePortServiceChanged(current, expected *corev1.Service) (*corev1.Service,
 	if len(current.Spec.Ports) != len(expected.Spec.Ports) {
 		updated.Spec.Ports = expected.Spec.Ports
 		changed = true
+		return updated, changed
 	}
 
 	for i, p := range current.Spec.Ports {
```

Comment:
to fit into the structure of this function, with only 1 location for a return, might be better to do an `else` case for the check for ranging through the list of ports, rather than return early